devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+		}
+
+		valueTile("heatingSetpoint", "device.heatingSetpoint", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "heat", label:'${currentValue} heat', unit: "F", backgroundColor:"#ffffff"
+		}
+		standardTile("heatDown", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "heatDown", label:'down', action:"heatDown", defaultState: true , icon:"st.thermostat.thermostat-down"
+		}
+		standardTile("heatUp", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "heatUp", label:'up', action:"heatUp", defaultState: true , icon:"st.thermostat.thermostat-up"
+		}
+
+		valueTile("coolingSetpoint", "device.coolingSetpoint", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "cool", label:'${currentValue} cool', unit:"F", backgroundColor:"#ffffff"
+		}
+		standardTile("coolDown", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "coolDown", label:'down', action:"coolDown", defaultState: true , icon:"st.thermostat.thermostat-down"
+		}
+		standardTile("coolUp", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "coolUp", label:'up', action:"coolUp", defaultState: true , icon:"st.thermostat.thermostat-up"
+		}
+
+		standardTile("mode", "device.thermostatMode", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "updating", label:'wait', action:"", backgroundColor:"#880000", nextState:"off" //, icon: "st.secondary.secondary"
+			state "off", action:"thermostat.auto", backgroundColor:"#ffffff", nextState:"updating" , icon: "st.thermostat.heating-cooling-off"
+			state "auto",action:"thermostat.cool", backgroundColor:"#00A0DC", nextState:"updating", icon: "st.thermostat.auto"
+			state "cool", action:"thermostat.heat", backgroundColor:"#00A0DC", nextState:"updating", icon: "st.thermostat.cool"
+			state "heat", action:"thermostat.off", backgroundColor:"#e86d13", nextState:"updating" , icon: "st.thermostat.heat"
+			
+			
+		}
+		standardTile("fanMode", "device.thermostatFanMode", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "updating", label:'wait', action:"", backgroundColor:"#ffffff", nextState:"fanAuto" // , icon: "st.secondary.secondary"
+			state "fanAuto",        action:"thermostat.fanOn", backgroundColor:"#ffffff" , nextState:"updating" , icon: "st.thermostat.fan-auto"
+			state "fanOn"		,   action:"thermostat.fanCirculate", backgroundColor:"#ffffff" ,  nextState:"updating" , icon: "st.thermostat.fan-on"
+			state "fanCirculate",   action:"thermostat.fanAuto", backgroundColor:"#ffffff" , nextState:"updating", icon: "st.thermostat.fan-circulate"
+			state "fanCirculate50", action:"thermostat.fanAuto", backgroundColor:"#ffffff" , nextState:"updating", icon: "st.thermostat.fan-circulate"
```

Comment:
consider adding a label to these additional states to distinguish them from the circulate state as the all use the same icon, e.g. label:'50'  or label:'circulate 50'

Reply:
The circulate icon comes with its own text super imposed on the label.  This not a real issue since only one of the circulate options is available on this UI (auto, on, circulate).  The other states are present in case the user has set it physically the device.  Cycling through 7-states seems unreasonable for one button.

Reply:
If you can't change the state in the DTH and the icon for the other circulate state is the same then there's no difference for the user so instead of listing several states here, map those that has to be set physically on the device to the circulate sate instead.